### PR TITLE
Renames and disentangles checksum validator classes.

### LINKED ISSUE #2068: Rename Audit::Checksum and/or ChecksumValidator
To make clearer that they are doing and how they are different.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #2068

## Why was this change made? 🤔
Less confusion, more comprehension.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, or if it changes code used internally for cloud replication, ***run [integration test preassembly_image_accessioning_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test/blob/main/spec/features/preassembly_image_accessioning_spec.rb) against stage as it tests preservation***, and/or test in stage environment, in addition to specs. The main classes relevant to replication are `ZipmakerJob`, `PlexerJob`, `*DeliveryJob`, `ResultsRecorderJob`, and `DruidVersionZip`; [see here for overview diagram of replication pipeline](https://github.com/sul-dlss/preservation_catalog/blob/main/app/jobs/README.md).⚡

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -79,15 +79,7 @@ RSpec/NamedSubject:
...
-  Max: 4 # default: 3
-  Exclude:
-    - 'spec/jobs/validate_moab_job_spec.rb'
-    - 'spec/services/audit/catalog_to_moab_spec.rb'
-    - 'spec/services/audit/moab_to_catalog_spec.rb'
-    - 'spec/requests/objects_controller_content_diff_spec.rb'
-    - 'spec/requests/objects_controller_file_spec.rb'
-    - 'spec/services/checksum_validator_spec.rb'
-    - 'spec/services/complete_moab_service/*.rb'
+  Enabled: false
```
Comment (justinlittman) [inline .rubocop.yml:82]: We just ignore this, so might as well make it false.

File: app/services/audit/checksum_validator.rb [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+      results.remove_db_updated_results unless transaction_ok
+      AuditResultsReporter.report_results(audit_results: results, logger: logger)
+    end
+
+    def object_dir
+      @object_dir ||= MoabOnStorage.object_dir(storage_location: storage_location, druid: druid)
+    end
+
+    def logger
+      @logger ||= Logger.new($stdout).extend(ActiveSupport::Logger.broadcast(Logger.new(Rails.root.join('log', 'cv.log'))))
```
Comment (justinlittman) [inline app/services/audit/checksum_validator.rb:111]: Gets its own logger, instead of getting from ChecksumValidatorUtils for less entanglement.

Comment (ndushay) [reply app/services/audit/checksum_validator.rb:111]: how about renaming the log to checksum_validator (and ensure log rotation and all that)

also make sure that change is propagated to any log rotation in qa, stage and prod environments

Comment (ndushay) [reply app/services/audit/checksum_validator.rb:111]: or maybe making a separate ticket for that?

**Review by ndushay — APPROVED** [review]
approving;  made a suggestion for your consideration;  leaving in case someone else wants to review and learn.

File: app/services/audit/checksum_validator.rb (hunk 2 above)
Comment (mjgiarlo) [reply app/services/audit/checksum_validator.rb:111]: @ndushay Any `*.log` files in the app `log/` directory should automatically be rotated. So we're clear to keep this name or change it w/o any need for further work or ticketing, AFAICT.

**Review by mjgiarlo — APPROVED** [review]
LGTM

**Pull request merged**